haxe/ui/components/Button.hx

```diff
@@ -184,9 +184,9 @@ class ButtonLayout extends DefaultLayout {
         var textAlign = cast(component, Button).textAlign;
         
         if (icon != null && (iconPosition == "far-right" || iconPosition == "far-left" || iconPosition == "center-left" || iconPosition == "center-right") && textAlign =="center") {
-            size.width -= icon.width * 2 + verticalSpacing;
-        } else if (icon != null && (iconPosition == "far-right" || iconPosition == "far-left" || icon != null && (iconPosition == "left" || iconPosition == "right" || iconPosition == "center-left" || iconPosition == "center-right"))) {
-            size.width -= icon.width + verticalSpacing;
+            size.width -= icon.width * 2 + horizontalSpacing;
```

Comment:
* for the extra space in the other side * -> this is need or else the text will not be centered right

size.width -= icon.width * 2 + horizontalSpacing * 2;